lib/ansible/modules/cloud/amazon/ec2_instance.py

```diff
@@ -75,65 +85,123 @@
     description:
       - An image to use for the instance. The M(ec2_ami_info) module may be used to retrieve images.
         One of I(image) or I(image_id) are required when instance is not already present.
-      - Complex object containing I(image.id), I(image.ramdisk), and I(image.kernel).
-      - I(image.id) is the AMI ID.
-      - I(image.ramdisk) overrides the AMI's default ramdisk ID.
-      - I(image.kernel) is a string AKI to override the AMI kernel.
+    type: dict
+    suboptions:
+      id:
+        description:
+        - The AMI ID.
+        type: str
+      ramdisk:
+        description:
+        - Overrides the AMI's default ramdisk ID.
+        type: str
+      kernel:
+        description:
+        - a string AKI to override the AMI kernel.
   image_id:
     description:
        - I(ami) ID to use for the instance. One of I(image) or I(image_id) are required when instance is not already present.
        - This is an alias for I(image.id).
+    type: str
   security_groups:
     description:
       - A list of security group IDs or names (strings). Mutually exclusive with I(security_group).
+    type: list
   security_group:
     description:
       - A security group ID or name. Mutually exclusive with I(security_groups).
+    type: str
   name:
     description:
       - The Name tag for the instance.
+    type: str
   vpc_subnet_id:
     description:
       - The subnet ID in which to launch the instance (VPC)
         If none is provided, ec2_instance will chose the default zone of the default VPC.
     aliases: ['subnet_id']
+    type: str
   network:
     description:
       - Either a dictionary containing the key 'interfaces' corresponding to a list of network interface IDs or
         containing specifications for a single network interface.
-      - If specifications for a single network are given, accepted keys are assign_public_ip (bool),
-        private_ip_address (str), ipv6_addresses (list), source_dest_check (bool), description (str),
-        delete_on_termination (bool), device_index (int), groups (list of security group IDs),
-        private_ip_addresses (list), subnet_id (str).
-      - I(network.interfaces) should be a list of ENI IDs (strings) or a list of objects containing the key I(id).
-      - Use the ec2_eni to create ENIs with special settings.
+      - Use the ec2_eni module to create ENIs with special settings.
+    type: dict
+    suboptions:
+      interfaces:
```

Comment:
It seems like a few things are now missing from this list - delete_on_termination, device_index, groups.